### PR TITLE
tests: indexer application registration addition and fix record view post permissions

### PR DESCRIPTION
**Pull request by hachreak**
Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

**Comment by hachreak** [comment]
ping @jirikuncar

**Comment by hachreak** [comment]
@jirikuncar  please, can you manually restart travis? there was a ES connection timeout

File: setup.cfg [hunk 1]
```diff
@@ -22,13 +22,13 @@
...
-test = pytest
-
 [build_sphinx]
 source-dir = docs/
 build-dir = docs/_build
 all_files = 1
 
 [bdist_wheel]
 universal = 1
+
```
Comment (jirikuncar) [inline setup.cfg]: why?

File: .gitignore [hunk 2]
```diff
@@ -59,3 +59,4 @@ target/
 # Examples instance directories
 examples/*_instance
 examples/*.db
+*.db
```
Comment (jirikuncar) [inline .gitignore]: Is it really needed? Can we make tests to clean their mess?

File: tests/conftest.py [hunk 3]
```diff
@@ -201,6 +202,7 @@ def es(app):
 @pytest.yield_fixture()
 def indexer(app, es):
     """Create a record indexer."""
+    InvenioIndexer(app)
```
Comment (jirikuncar) [inline tests/conftest.py]: 👍

File: setup.cfg (hunk 1 above)
Comment (hachreak) [inline setup.cfg]: for a unknown reason, the python plugin for vim stop to work with aliases as first.. @.@

**Comment by jirikuncar** [comment]
I have moved minimal changes to #69. Please refactor conftest.py and test_permissions.py to avoid code duplication.

File: tests/conftest.py [hunk 4]
```diff
@@ -40,16 +41,47 @@
...
+
+def deny_login(*args, **kwargs):
+    """Return permission that deny access simulating required_login."""
+    if 'record' in kwargs and kwargs['record']:
+        return type('Deny', (), {'can': lambda self: True})()
+    else:
+        return type('Allow', (), {'can': lambda self: False})()
+
+
+class IndexingRecord(Record):
```
Comment (jirikuncar) [inline tests/conftest.py]: consider using `indexer` fixture explicitly instead of overriding `create` method.

File: tests/conftest.py [hunk 5]
```diff
@@ -40,16 +41,47 @@
...
 from invenio_search import InvenioSearch, RecordsSearch, current_search, \
     current_search_client
 from sqlalchemy_utils.functions import create_database, database_exists
 
 from invenio_records_rest import InvenioRecordsREST, config
 from invenio_records_rest.facets import terms_filter
+from invenio_records_rest.utils import allow_all, deny_all
+
+
+def deny_record(*args, **kwargs):
```
Comment (jirikuncar) [inline tests/conftest.py]:
```python
def deny_record(*args, **kwargs):
    """Return permission that always deny an access."""
    return type('Allow', (), {
        'can': lambda self: not ('record' in kwargs and kwargs['record'])
    })()
```

File: tests/conftest.py (hunk 4 above)
Comment (hachreak) [inline tests/conftest.py]: there is a precise reason if I need to do that.
I need to be sure that if the permission is denied, nothing is written on elasticsearch (as is happened before this PR on [invenio-deposit](https://github.com/inveniosoftware/invenio-deposit/blob/master/invenio_deposit/api.py#L137))

Comment (jirikuncar) [inline tests/conftest.py]: Permission checking using ES is not a core functionality of this module. We are trying to keep the tests simple.

**Comment by lnielsen** [comment]
@hachreak Will you update and rebase this PR.

**Comment by hachreak** [comment]
closed by https://github.com/inveniosoftware/invenio-records-rest/commit/aaf469946242eea995df0e9896539376d3db762f

**Pull request closed without merging**